geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java

```diff
@@ -1384,153 +1323,145 @@ protected void txApplyEntryOpAdjunctOnly(FarSideEntryOp entryOp) {
+          msg.addFarSideEntryOp(entryOp);
+          opKeys.add(entryOp.key);
+          opEntries.add(entryOp);
         }
       }
     }
 
     @Override
     public String toString() {
       StringBuilder result = new StringBuilder(64);
-      if (this.regionPath != null) {
-        result.append(this.regionPath);
+      if (regionPath != null) {
+        result.append(regionPath);
       } else {
-        result.append(this.internalRegion.getFullPath());
+        result.append(internalRegion.getFullPath());
       }
-      if (this.refCount > 0) {
-        result.append(" refCount=").append(this.refCount);
+      if (refCount > 0) {
+        result.append(" refCount=").append(refCount);
       }
       return result.toString();
     }
 
     private void basicToData(DataOutput out,
         SerializationContext context,
         boolean useShadowKey) throws IOException {
-      if (this.internalRegion != null) {
-        DataSerializer.writeString(this.internalRegion.getFullPath(), out);
-        if (this.internalRegion instanceof BucketRegion) {
+      if (internalRegion != null) {
+        DataSerializer.writeString(internalRegion.getFullPath(), out);
+        if (internalRegion instanceof BucketRegion) {
           DataSerializer.writeString(
-              ((Bucket) this.internalRegion).getPartitionedRegion().getFullPath(), out);
+              ((Bucket) internalRegion).getPartitionedRegion().getFullPath(), out);
         } else {
           DataSerializer.writeString(null, out);
         }
       } else {
-        DataSerializer.writeString(this.regionPath, out);
-        DataSerializer.writeString(this.parentRegionPath, out);
+        DataSerializer.writeString(regionPath, out);
+        DataSerializer.writeString(parentRegionPath, out);
       }
 
-      if (isEmpty() || this.opKeys.size() == 0) {
+      if (isEmpty() || opKeys.size() == 0) {
         out.writeInt(0);
       } else {
-        int size = this.opKeys.size();
+        int size = opKeys.size();
         out.writeInt(size);
 
         final boolean largeModCount;
-        if (this.msg.txState != null) {
-          largeModCount = this.msg.txState.needsLargeModCount();
+        if (msg.txState != null) {
+          largeModCount = msg.txState.needsLargeModCount();
         } else {
-          largeModCount = this.msg.needsLargeModCount;
+          largeModCount = msg.needsLargeModCount;
         }
         out.writeBoolean(largeModCount);
 
-        final boolean sendVersionTags =
-            this.msg.clientVersion == null
-                || KnownVersion.GFE_70.compareTo(this.msg.clientVersion) <= 0;
-        if (sendVersionTags) {
-          VersionSource member = this.memberId;
-          if (member == null) {
-            if (this.internalRegion == null) {
-              Assert.assertTrue(this.msg.txState == null);
-            } else {
-              member = this.internalRegion.getVersionMember();
-            }
+        VersionSource member = memberId;
```

Comment:
Here we have completely removed the check for msg.clientVersion for null. Looking at the code there are some places where this valus is set to nul like in CommitCommand. Now the member version source is always sent, is this behavior change correct?

Reply:
It would send the `VersionSource` if it didn't know the `clientVersion` or it was newer than v7.0. In this case the latter part is always true so the check for unknown or `null` `clientVersion` is redundant.

Reply:
Thank you for the explanation